daemon/algod/api/server/v2/handlers.go

```diff
@@ -267,22 +267,15 @@ func (v2 *Handlers) generateKeyHandler(address string, params model.GeneratePart
 // GenerateParticipationKeys generates and installs participation keys to the node.
 // (POST /v2/participation/generate/{address})
 func (v2 *Handlers) GenerateParticipationKeys(ctx echo.Context, address string, params model.GenerateParticipationKeysParams) error {
-	if v2.Keygen == nil {
-		v2.Keygen = make(chan struct{}, 1)
-	}
-
-	select {
-	case v2.Keygen <- struct{}{}:
+	if v2.KeygenLimiter != nil && v2.KeygenLimiter.TryAcquire(1) {
```

Comment:
```suggestion
	if v2.KeygenLimiter.TryAcquire(1) {
```

I think I'd like to avoid the `nil` check, as it implies `nil` is an ok thing. We're always initializing now,right? I think a `nil` here is a panic worthy event.

Also, lovers of outdentation might prefer putting the `!TryAcquire(1)` case first, and then not needing the `else`.

Reply:
I dunno, one of the proverbs is "[Make the zero value useful.](https://go-proverbs.github.io/)"

I was also imagining a world where we had more tests, so this would make it easier to properly create the handler object.

Reply:
I don't feel that strongly, but
1) The zero value would still be useful (unless you called this function, which would be bad!)
2) Because of that, tests can also elide initializing it, just as now.

Reply:
I think it's not even very useful the way I have it here. If it wasn't set and you call this function the feature is effectively disabled and the error message is wrong.